### PR TITLE
Passing trace context to HTTP output binding in metadata

### PR DESCRIPTION
**Pull request by shalabhms**
# Description
Passing trace context to http output binding in metadata.

## Issue reference
Closes https://github.com/dapr/dapr/issues/1913
https://github.com/dapr/components-contrib/issues/424

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/http/api.go [hunk 1]
```diff
@@ -302,6 +305,15 @@ func (a *api) onOutputBindingMessage(reqCtx *fasthttp.RequestCtx) {
...
 
+	// pass the trace context to output binding in metadata
+	if span := diag_utils.SpanFromContext(reqCtx); span != nil {
+		sc := span.SpanContext()
+		metadata := map[string]string{}
+		metadata[traceparentHeader] = diag.SpanContextToW3CString(sc)
+		metadata[tracestateHeader] = diag.TraceStateToW3CString(sc)
+		req.Metadata = metadata
+	}
+
```
Comment (youngbupark) [inline pkg/http/api.go:316]: Can you add test? at least unit-test.

Comment (shalabhms) [reply pkg/http/api.go:316]: The unit test for this method with tracing is there and this code path will be taken care of this test ..as no failure observed.

https://github.com/dapr/dapr/blob/613398ca4cf269741dd0957425adb01b0f965aec/pkg/http/api_test.go#L113

**Pull request merged**